components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML1BearerGrantHandler.java

```diff
@@ -0,0 +1,516 @@
+import org.opensaml.xml.signature.SignatureValidator;
+import org.opensaml.xml.validation.ValidationException;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.wso2.carbon.base.MultitenantConstants;
+import org.wso2.carbon.identity.application.common.IdentityApplicationManagementException;
+import org.wso2.carbon.identity.application.common.model.FederatedAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.model.IdentityProvider;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants;
+import org.wso2.carbon.identity.application.common.util.IdentityApplicationManagementUtil;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
+import org.wso2.carbon.identity.oauth.internal.OAuthComponentServiceHolder;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.model.RequestParameter;
+import org.wso2.carbon.identity.oauth2.token.OAuthTokenReqMessageContext;
+import org.wso2.carbon.identity.oauth2.token.handlers.grant.AbstractAuthorizationGrantHandler;
+import org.wso2.carbon.identity.oauth2.token.handlers.grant.saml.SAML2TokenCallbackHandler;
+import org.wso2.carbon.identity.oauth2.util.X509CredentialImpl;
+import org.wso2.carbon.idp.mgt.IdentityProviderManager;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.service.RealmService;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.*;
+
+/**
+ * This implements SAML 1.0 Bearer Assertion Profile(this is to be documented) for OAuth 2.0
+ */
+public class SAML1BearerGrantHandler extends AbstractAuthorizationGrantHandler {
+
+    private static Log log = LogFactory.getLog(SAML1BearerGrantHandler.class);
+
+    SAMLSignatureProfileValidator profileValidator = null;
+    private boolean audienceRestrictionValidationEnabled = false;
+    public static final String OAUTH_SAML_1_BEARER_METHOD = "urn:oasis:names:tc:SAML:1.0:cm:bearer";
+    private static final String SAML10_BEARER_GRANT_TYPE_CONFIG_FILE="SAML10_BearerGrantType.properties";
+    public void init() throws IdentityOAuth2Exception {
+
+        super.init();
+
+        Thread thread = Thread.currentThread();
+        ClassLoader loader = thread.getContextClassLoader();
+        thread.setContextClassLoader(this.getClass().getClassLoader());
+
+        try {
+            DefaultBootstrap.bootstrap();
+        } catch (ConfigurationException e) {
+            log.error(e.getMessage(),e);
+            throw new IdentityOAuth2Exception("Error in bootstrapping the OpenSAML2 library");
```

Comment:
Should include 'e' when throwing exception?
Like "throw new IdentityOAuth2Exception("Error in bootstrapping the OpenSAML2 library",e);"